t/mock.t

```diff
@@ -1787,3 +1792,46 @@ GET /t
 records: [{"class":1,"expire":1800,"minimum":60,"mname":"ns3.google.com","name":"google.com","refresh":900,"retry":900,"rname":"dns-admin.google.com","section":1,"serial":175802026,"ttl":0,"type":6},{"address":"127.0.0.1","class":1,"name":"ns3.google.com","section":3,"ttl":0,"type":1}]
 --- no_error_log
 [error]
+
+
+
+=== TEST 36: retry on connection failures
+--- http_config eval: $::HttpConfig
+--- config
+    location /t {
+        content_by_lua '
+            local resolver = require "resty.dns.resolver"
+
+            local r, err = resolver:new{
+                nameservers = { {"127.0.0.1", 20000} },  -- note: bad port
```

Comment:
For the sake of completeness, will you add another test case for retrying across multiple nameservers?